lib/ace/mode/csp_highlight_rules.js

```diff
@@ -0,0 +1,214 @@
+                    comment: "https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/plugin-types"
+                },
+                {
+                    token: "constant.language.character",
+                    regex: /sandbox/,
+                    comment: "https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/sandbox"
+                },
+                {
+                    token: "constant.language.character",
+                    regex: /disown-opener/,
+                    comment: "https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/disown-opener"
+                },
+                {
+                    token: "constant.language.character",
+                    regex: /form-action/,
+                    comment: "https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/form-action"
+                },
+                {
+                    token: "constant.language.character",
+                    regex: /frame-ancestors/,
+                    comment: "https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-ancestors"
+                },
+                {
+                    token: "constant.language.character",
+                    regex: /report-uri/,
+                    comment: "https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/report-uri"
+                },
+                {
+                    token: "constant.language.character",
+                    regex: /report-to/,
+                    comment: "https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/report-to"
+                },
+                {
+                    token: "constant.language.character",
+                    regex: /upgrade-insecure-requests/,
+                    comment: "https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/upgrade-insecure-requests"
+                },
+                {
+                    token: "constant.language.character",
+                    regex: /block-all-mixed-content/,
+                    comment: "https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/block-all-mixed-content"
+                },
+                {
+                    token: "constant.language.character",
+                    regex: /require-sri-for/,
+                    comment: "https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/require-sri-for"
+                },
+                {
+                    token: "constant.language.character",
+                    regex: /reflected-xss/,
+                    comment: "https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/reflected-xss"
+                },
+                {
+                    token: "constant.language.character",
+                    regex: /referrer/,
```

Comment:
it would be better  to group these into one regex or use a [keywordMapper](https://github.com/ajaxorg/ace/blob/master/lib/ace/mode/javascript_highlight_rules.js#L43)

Reply:
@nightwing I'll explain my reasoning and you tell me if it is broken :) from the documentation I understood that the matching will stop once something is found, so there is no need traversing each rule. Thus less work for the editor. If I mush everything in one regex, won't that be more taxing for the IDE? If that is not the case I'll change it.

Reply:
Using one regexp is faster than using several, so highlighter combines all the rules in the state together https://github.com/ajaxorg/ace/blob/2d768f6f4f4bca1022ef12ebacf5c1396aaed9a4/lib/ace/tokenizer.js#L131.
Here it still needs to see which rule did match, and that adds some overhead.